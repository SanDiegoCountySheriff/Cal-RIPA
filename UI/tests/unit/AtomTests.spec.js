// import { createLocalVue, mount } from '@vue/test-utils'
// import RipaLabel from './../../src/components/atoms/RipaLabel.vue'
// import RipaAlert from './../../src/components/atoms/RipaAlert.vue'
// import RipaSelect from './../../src/components/atoms/RipaSelect.vue'
// import Vue from 'vue'
// import Vuetify from 'vuetify'

// Vue.use(Vuetify)

// describe('Ripa Label Tests', () => {
//   it('should have a bold label', async () => {
//     const wrapper = mount(RipaLabel, {
//       propsData: {
//         value: 'Label',
//         bold: true,
//       },
//     })

//     const div = wrapper.find('div')
//     const span = wrapper.find('span')

//     expect(div.classes().includes('tw-font-bold')).toBe(true)
//     expect(wrapper.vm.boldClass).toBe('tw-font-bold')
//     expect(span.text()).toBe('Label')

//     await wrapper.setProps({ value: 'New Label', bold: false })

//     expect(div.classes().includes('tw-font-bold')).toBe(false)
//     expect(wrapper.vm.boldClass).toBe('')
//     expect(span.text()).toBe('New Label')
//   })
// })

// describe('Ripa Alert tests', () => {
//   const localVue = createLocalVue()
//   let vuetify

//   beforeEach(() => {
//     vuetify = new Vuetify()
//   })

//   it('should outline correctly', async () => {
//     const wrapper = mount(RipaAlert, {
//       localVue,
//       vuetify,
//       propsData: {
//         alertType: 'warning',
//         alertOutlined: true,
//       },
//     })

//     const topDiv = wrapper.find('div')

//     expect(topDiv.classes().includes('warning--text')).toBe(true)
//     expect(topDiv.classes().includes('v-alert--outlined')).toBe(true)

//     await wrapper.setProps({ alertOutlined: false, alertType: 'info' })

//     expect(topDiv.classes().includes('warning--text')).toBe(false)
//     expect(topDiv.classes().includes('v-alert--outlined')).toBe(false)
//   })
// })

// describe('Ripa Select Tests', () => {
//   const localVue = createLocalVue()
//   let vuetify

//   beforeEach(() => {
//     vuetify = new Vuetify()
//   })

//   it('should render correctly', () => {
//     const value = ['test', 1]
//     const wrapper = mount(RipaSelect, {
//       localVue,
//       vuetify,
//       propsData: {
//         value: value,
//       },
//     })
//     expect(wrapper.element).toMatchSnapshot()
//   })
// })
