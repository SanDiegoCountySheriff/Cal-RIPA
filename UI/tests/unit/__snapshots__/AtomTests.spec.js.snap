// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Select Tests should render correctly 1`] = `
<div
  class="v-input theme--light v-text-field v-text-field--solo-flat v-select"
>
  <div
    class="v-input__control"
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-owns="list-8"
      class="v-input__slot"
      role="button"
    >
      <div
        class="v-select__slot"
      >
        <div
          class="v-select__selections"
        >
          <input
            aria-readonly="false"
            autocomplete="off"
            clerable=""
            id="input-8"
            readonly="readonly"
            required="required"
            type="text"
          />
        </div>
        <div
          class="v-input__append-inner"
        >
          <div
            class="v-input__icon v-input__icon--append"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-menu-down theme--light"
            />
          </div>
        </div>
        <input
          type="hidden"
          value="test,1"
        />
      </div>
      <div
        class="v-menu"
      >
        <!---->
      </div>
    </div>
    <div
      class="v-text-field__details"
    >
      <div
        class="v-messages theme--light"
      >
        <transition-group-stub
          class="v-messages__wrapper"
          name="message-transition"
          tag="div"
        />
      </div>
    </div>
  </div>
</div>
`;
